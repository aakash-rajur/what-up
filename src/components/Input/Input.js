import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Input extends Component {
    static propTypes = {
        onValueReturn: PropTypes.func
    };

    constructor(props) {
        super(props);
        this.onReturn = this.onReturn.bind(this);
    }

    render() {
        let inputProps = {...this.props};
        delete inputProps.onValueReturn;
        return (
            <input {...inputProps} onKeyUp={this.onReturn}/>
        )
    }

    onReturn(e) {
        let {
            keyCode,
            target: {value}
        } = e, {
            onKeyUp,
            onValueReturn
        } = this.props;
        onKeyUp && onKeyUp(e);
        keyCode === 13 && onValueReturn && onValueReturn(value)
    }
}

export default Input;
