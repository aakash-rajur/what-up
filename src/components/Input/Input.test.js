import React from 'react';
import Input from './Input';

describe('Input component', () => {
	let props = {
		type: 'text', tabIndex: 1, className: 'task-input',
		autoFocus: true, value: 'hello world',
		placeholder: 'lorem ipsum', onValueReturn: stub()
	}, component = shallow(<Input {...props}/>);
	
	it('should render', () => {
		expect(component.find('input').exists())
			.toBe(true);
	});
	
	it('props integrity', () => {
		let domProps = {...props};
		delete domProps.onValueReturn;
		Object.entries(domProps)
			.forEach(([prop, value]) =>
				expect(component.prop(prop))
					.toBe(value)
			);
	});
	
	it('should not fire onValueReturn if keyCode is not 13', () => {
		component.simulate('keyup', {keyCode: 10, target: {value: props.value}});
		expect(props.onValueReturn.called).toBe(false);
	});
	
	it('should fire onValueReturn if keyCode is 13', () => {
		component.simulate('keyup', {keyCode: 13, target: {value: props.value}});
		expect(props.onValueReturn.called).toBe(true);
		expect(props.onValueReturn.callCount).toBe(1);
		expect(props.onValueReturn.calledWithExactly(props.value)).toBe(true)
	});
	
	it('match snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});
