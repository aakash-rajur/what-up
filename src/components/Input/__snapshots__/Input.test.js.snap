// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component match snapshot 1`] = `
<input
  autoFocus={true}
  className="task-input"
  onKeyUp={[Function]}
  placeholder="lorem ipsum"
  tabIndex={1}
  type="text"
  value="hello world"
/>
`;
